cmd/manager/main.go

```diff
@@ -225,3 +247,90 @@ func checkIfRequiredFilesExist(requiredFiles []string) error {
 	}
 	return nil
 }
+
+// checkConfiguration checks for required configuration in a cluster before starting the controller.
+func (c *clusterConfig) checkConfiguration(config *rest.Config) error {
+	// check OpenShift API config client.
+	if err := c.getOpenShiftConfigClient(config); err != nil {
+		return fmt.Errorf("unable to get OpenShift client: %v", err)
+	}
+	// check OpenShift API operator client
+	if err := c.getOpenShiftOperatorClient(config); err != nil {
+		return fmt.Errorf("unable to get OpenShift operator client: %v", err)
+	}
```

Comment:
These sort of checks are typically when creating the object, like `newClusterConfig()`. Think about a scenario where we will be adding more methods to `clusterConfig` struct that use these clients.